src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -254,10 +256,14 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
 
                 Debug.Assert(Encoding.ASCII.GetString(_hostHeaderLineBytes) == $"Host: {hostHeader}\r\n");
 
-                if (sslHostName == null)
+                if (sslHostName == null && _http2Enabled)
                 {
                     _http2EncodedAuthorityHostHeader = HPackEncoder.EncodeLiteralHeaderFieldWithoutIndexingToAllocatedArray(H2StaticTable.Authority, hostHeader);
-                    _http3EncodedAuthorityHostHeader = QPackEncoder.EncodeLiteralHeaderFieldWithStaticNameReferenceToArray(H3StaticTable.Authority, hostHeader);
+
+                    if (IsHttp3Supported() && _http3Enabled)
```

Comment:
In what situation would http/3 be enabled but not supported?

Reply:
None, but we guard H3-specific logic like this in other places as well (I assume to let it be trimmed out)

Reply:
> to let it be trimmed

Ah, right